src/broker/amqp.rs

```diff
@@ -301,13 +311,21 @@ impl Broker for AMQPBroker {
     }
 
     async fn close(&self) -> Result<(), BrokerError> {
-        // 320 reply-code = "connection-forced", operator intervened.
-        // For reference see https://www.rabbitmq.com/amqp-0-9-1-reference.html#domain.reply-code
+        let consume_channel = self.consume_channel.write().await;
+        let produce_channel = self.produce_channel.write().await;
         let conn = self.conn.lock().await;
+
         if conn.status().connected() {
+            // 320 reply-code = "connection-forced", operator intervened.
+            // For reference see https://www.rabbitmq.com/amqp-0-9-1-reference.html#domain.reply-code
+            debug!("Closing consumer channel...");
+            consume_channel.close(320, "").await?;
```

Comment:
@Keruspe this is where the error arises.

```
[ERROR lapin::channels] Connection error: invalid channel state: Closing
```

It's as if the channel already been closed. But I'm not sure how that's possible.

Reply:
Weirdly, if you close the producer channel first, there is no error.